Mathlib/AlgebraicTopology/ModelCategory/JoyalTrick.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.AlgebraicTopology.ModelCategory.CategoryWithCofibrations
+import Mathlib.CategoryTheory.MorphismProperty.Limits
+
+/-!
+# A trick by Joyal
+
+In order to construct a model category, we may sometimes have basically
+proven all the axioms with the exception of the left lifting property
+of cofibrations with respect to trivial fibrations. A trick by Joyal
+allows to obtain this lifting property under suitable assumptions,
+namely that cofibrations are stable under composition and cobase change.
+(The dual result is also formalized.)
+
+## References
+* [John F. Jardine, Simplicial presheaves][jardine-1987]
+
+-/
+
+open CategoryTheory Category Limits MorphismProperty
+
+namespace HomotopicalAlgebra
+
+namespace ModelCategory
+
+variable {C : Type*} [Category C]
+  [CategoryWithCofibrations C] [CategoryWithFibrations C] [CategoryWithWeakEquivalences C]
+  [(weakEquivalences C).HasTwoOutOfThreeProperty]
+
+lemma joyal_trick
+    [HasFactorization (cofibrations C) (trivialFibrations C)] [HasPushouts C]
+    [(cofibrations C).IsStableUnderComposition] [(cofibrations C).IsStableUnderCobaseChange]
+    (h : ∀ {A B X Y : C} (i : A ⟶ B) (p : X ⟶ Y)
+      [Cofibration i] [WeakEquivalence i] [Fibration p], HasLiftingProperty i p)
+    {A B X Y : C} (i : A ⟶ B) (p : X ⟶ Y)
+    [Cofibration i] [Fibration p] [WeakEquivalence p] :
+    HasLiftingProperty i p where
+  sq_hasLift {f g} sq := by
+    have h := factorizationData (cofibrations C) (trivialFibrations C)
+      (pushout.desc p g sq.w)
+    have sq' : CommSq (𝟙 X) (pushout.inl _ _ ≫ h.i) p h.p := ⟨by simp⟩
```

Comment:
```suggestion
    have sq' : CommSq (𝟙 X) (pushout.inl _ _ ≫ h.i) p h.p := .mk
```

I guess we have the autoparam now, so this is possible (or a `{ }` instead of ̀`.mk` but this looks a bit weird).